iris-mpc-cpu/src/hawkers/galois_store.rs

```diff
@@ -166,7 +157,7 @@ async fn eval_pairwise_distances(
 impl VectorStore for LocalNetAby3NgStoreProtocol {
     type QueryRef = PointId; // Vector ID, pending insertion.
     type VectorRef = PointId; // Vector ID, inserted.
-    type DistanceRef = Vec<DistanceShare<u16>>; // Distance represented as shares.
+    type DistanceRef = Vec<AuthDistanceShare<u16>>; // Distance represented as shares.
```

Comment:
I also think the current construction is kind of leaking the locality of the current non-networked implementation.
I would rather, like we did for our inititial CPU iris prototypes have a protocol that is generic over the networking, where the networking can then just be handled by channels as it is now for the local variant and by e.g. GRPC for the networked variant, but have the remaining protocol be the same. This would then mean that we do 3 instances of the local protocol for local testing in 3 different threads, rather than the current protocol having one instance and encapsulating this internally. IMO this is cleaner as otherwise some of the internal functionality is duplicated between these two versions?

Reply:
Valid points, guys. Let me then rewrite this local store to get rid of these player identities lurking in evaluation.

Reply:
@dkales btw, it's essentially what Dragos did here #618, but I agree that it doesn't make much sense to have a separate `VectorStore` implementation. It's rather better to refactor the local one here.

Reply:
Done. Please, have a look at the updated PR description